backends/grpc.go

```diff
@@ -165,13 +199,28 @@ func createClient(hostname string, caCert, tlsCert, tlsKey []byte) (*grpc.Client
 		})))
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
+	return nsOpts, nil
```

Comment:
What do you think about removing the [`grpc.WithBlock()`](https://pkg.go.dev/google.golang.org/grpc#WithBlock) (optional) from the options? This would remove the need to also open the connection for the `grpc.DialContext` call. Furthermore, you could still set a backoff configuration and GRPC would take care of the connection itself. (By default, I believe, a backoff strategy is defined. )

So, you could run `grpc.DialContext` once and GRPC will take care of the connection on its own and reconnect if the connection was interrupted / not established.